ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm.hpp

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+
+namespace ttnn::operations::moreh::moreh_clip_grad_norm {
+
+struct MorehClipGradNorm {
+    static Tensor invoke(
+        const std::vector<Tensor> &inputs,
+        float max_norm,
+        float norm_type,
+        bool error_if_nonfinite,
```

Comment:
Can we please replace this with enum class?
https://github.com/tenstorrent/tt-metal/blob/main/best_practices.md#14-avoid-bool-arguments-in-apis